department-of-veterans-affairs/caseflow
# Add generated schema information to Rails model files

Resolves having to search through `schema.rb` or https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/caseflow_schema_csv.html.

### Description
Add the [annotate gem](https://github.com/ctran/annotate_models) and add automatically generated annotations to model files only.

I find myself frequently searching through `schema.rb`; this would avoid or at least reduce the need to do that.

An added bonus is that indexed columns and foreign keys are more obvious.

### Acceptance Criteria
- [ ] Manually created comments are preserved after running `bundle exec annotate`

### Testing Plan
1. Add comments to model files
2. Run `bundle exec annotate`
3. Ensure your comments are not modified

### Review tip
Hide all `.rb` files to see the crux of this PR:
![image](https://user-images.githubusercontent.com/55255674/145070182-aa18188b-f0b2-45d4-9be8-74d4ff1891d2.png)

Then show only `.rb` files to check that no lines were removed:
![image](https://user-images.githubusercontent.com/55255674/145070522-69c61260-31f9-40f2-b965-52ec2c692bb9.png)

Also the PR shows no lines removed:
![image](https://user-images.githubusercontent.com/55255674/145075930-ec3131b9-ddd2-4d5e-8d7d-95995c9057b9.png)
